File: src/dotnet/Trivi/Trivi.WebGui/wwwroot/js/ts/custom/domain/helpers/new-questions/true-false.ts
import { QuestionType } from "../../enums/question-type";
import { PutTrueFalseApiRequest } from "../../models/question-models";
import { NewQuestion } from "./new-question";


export class NewTrueFalse extends NewQuestion<PutTrueFalseApiRequest>
{
    public readonly getQuestionType = () => QuestionType.TrueFalse;
    protected readonly _defaultPrompt = "New true false";

    public toForm(): PutTrueFalseApiRequest
    {
        return {
            collectionId: this._collectionId,
            correctAnswer: true,
            prompt: this._defaultPrompt,
        };
    }
}

File: src/dotnet/Trivi/Trivi.Lib/Domain/Models/TrueFalse.cs
using Trivi.Lib.Domain.Forms;
using Trivi.Lib.Domain.Other;

namespace Trivi.Lib.Domain.Models;

public class TrueFalse : Question
{
    public override QuestionType QuestionType => QuestionType.TrueFalse;

    public bool CorrectAnswer { get; set; } = true;


    public static TrueFalse FromRequestForm(QuestionId questionId, TrueFalseForm form)
    {
        return new()
        {
            Id = questionId,
            CollectionId = form.CollectionId,
            CorrectAnswer = form.CorrectAnswer,
            Prompt = form.Prompt,
            Points = form.Points,
        };
    }
}
